### ISSUE TITLE
UnitType Revit API Changes

### ISSUE THREAD

**Issue by jmcouffin** [eirannejad/pyRevit#1839]
fixed in DB module with :
```python
        if HOST_APP.is_newer_than(2022, or_equal=True):
            # GetSpecTypeId() Removed in Revit 2022
            self.unit_type = self.param_def.GetDataType()
        elif HOST_APP.is_exactly(2021):
            # Revit >2021 does not have the UnitType property
            self.unit_type = self.param_def.GetSpecTypeId()
        else:
            self.unit_type = self.param_def.UnitType
```

**Issues here:**

- List Elements - Data Schemas

> "C:\pyRevit\latest\latest\extensions\pyRevitCore.extension\pyRevit.tab\pyRevit.panel\tools.stack\Spy.pulldown\List Elements.pushbutton\script.py", line 163, in <module>
> AttributeError: 'Field' object has no attribute 'UnitType

- Calculate Average Area

> File "C:\pyRevit\latest\latest\extensions\pyRevitTools.extension\pyRevit.tab\Analysis.panel\Tools.stack\Analyse.pulldown\Calculate Average Area.pushbutton\script.py", line 82, in <module>
> File "C:\pyRevit\latest\latest\extensions\pyRevitTools.extension\pyRevit.tab\Analysis.panel\Tools.stack\Analyse.pulldown\Calculate Average Area.pushbutton\script.py", line 47, in calc_and_print
> File "C:\pyRevit\latest\latest\pyrevitlib\pyrevit\revit\units.py", line 18, in format_area
> AttributeError: 'Autodesk.Revit.DB' object has no attribute 'UnitType'

- Find Range of Roof Slopes

> "C:\pyRevit\latest\latest\extensions\pyRevitTools.extension\pyRevit.tab\Analysis.panel\Tools.stack\Analyse.pulldown\Find Range Of Roof Slopes.pushbutton\script.py", line 17, in <module>
>  File "C:\pyRevit\latest\latest\pyrevitlib\pyrevit\revit\units.py", line 37, in format_slope
> TypeError: Format() got an unexpected keyword argument 'unitType'
> Script Executor Traceback:
> Microsoft.Scripting.ArgumentTypeException: Format() got an unexpected keyword argument 'unitType'

quick search through repo code in .py files 
![image](https://github.com/eirannejad/pyRevit/assets/7872003/27737a9d-d844-4c01-bfe4-6fa860e3cf6c)

**Comment by jmcouffin** [comment]
ongoing list whiel testing

**Comment by eirannejad** [comment]
Fixed in develop up to here

**Comment by jmcouffin** [comment]
Fixed the wipe all project parameters with #1842 
What's funny is that the tool was not listed in the bundle layout!

That might help to fix the rest

```
    if HOST_APP.is_newer_than('2022'):
        ut = str(p.GetDataType().TypeId)
        tp = str(p.GetDataType().TypeId.split('.')[-3])
    elif HOST_APP.is_exactly('2022'):
        ut = str(p.GetSpecTypeId().TypeId)
        tp = str(p.ParameterType)
    else:
        ut = str(p.UnitType)
        tp = str(p.ParameterType)
    pg = str(p.ParameterGroup)
```

**Comment by jmcouffin** [comment]
I checked again this morning, nothing left that I can spot. I will close the issue.
